AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java

```diff
@@ -1205,10 +1208,35 @@ public int _revForDeck(long did, int lim) {
     	return mCol.getDb().queryScalar("SELECT count() FROM (SELECT 1 FROM cards WHERE did = " + did + " AND queue = 2 AND due <= " + mToday + " LIMIT " + lim + ")");
     }
 
-    public int _youngRevCountForDeck(long did) {
-        return mCol.getDb().queryScalar(
-            "SELECT COUNT(*) FROM cards WHERE did = " + did + " AND queue = 2 AND due <= " +
-                mToday + " AND ivl <= 21");
+    public int _criticalReviewCountForDecks(LinkedList<Long> dids) {
+        int total = 0;
+        for (long did : dids) {
+            total += _criticalReviewCountForDeck(did);
+        }
+        return total;
+    }
+
+    public int _criticalReviewCountForDeck(long did) {
+        return mCol.getDb().queryScalar(getCriticalReviewQueryGivenSelect("COUNT(*)", did));
+    }
+
+    public String getCriticalReviewQueryGivenSelect(String selectString, long did) {
+        // using DISTINCT because our join here is approximate (multiple revlog entries could match
```

Comment:
interested to hear more about how multiple revlog entries could match on all these criteria. when would ease factor/interval pair not change between multiple reviews of same card?

Reply:
Yeah it's not likely but could happen. I think when I ran the query 2 more cards matched without the distinct. Ease factor / interval can be the same if you eg do 10 good reviews, then get one wrong (-15% ivl), then do easy (+15% ivl). Eventually there might be an `ivl` match, since we've got the same `cid` and `factor` and our current `ivl` is somewhere between a bunch of `ivl`s already in the db. Ultimately not a big deal.